File: LeetCodeSolutions/LeetCode_0009.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        n = x
        if n < 0:
            return False
        if n == 0: return True
        if n % 10 == 0:
            return False
        tmp = 10
        org = n
        res = 0
        while n:
            res = res * 10 + n % tmp
            n //= tmp
        print(org, res)
        return org == res

File: LeetCodeSolutions/LeetCode_0009.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        if (x < 10) return true;
        if (x % 10 == 0) return false;
        int r = 0;
        while (x > r){
            int n = x % 10;
            x /= 10;
            r *= 10;
            r += n;
        }

        return r == x || r / 10 == x;
    }
};